### ISSUE TITLE
New user roles improvements

### ISSUE THREAD

**Issue by maxceem** [appirio-tech/connect-app#3345]
In the recent issue https://github.com/appirio-tech/connect-app/issues/3298 we introduced new user roles while keeping the logic the same as it is. We just made the new roles behaves same as old roles.

Now as we've released that part we can move forward and make the code behave depend on the role more specific.

What can be done:
1. Let only invite users with the exact roles (not like any manager user can be invited with any manager role). Related code https://github.com/topcoder-platform/tc-project-service/blob/feature/new-roles/src/routes/projectMemberInvites/create.js#L290-L307
2. Restrict users so they can join project using **Join** button, only with their role (not like any manager user can be invited with any manager role). See conversation on this PR https://github.com/topcoder-platform/tc-project-service/pull/387. Most likely it's already done on the Project Service with some additional permissions for some roles, see related code https://github.com/topcoder-platform/tc-project-service/blob/dev/src/routes/projectMembers/create.js#L43-L95. We may want to update these rules with a more consistent approach or some general logic here. We also have to restrict which roles managers can choose when joining the project themselves in the [joining dialog](https://user-images.githubusercontent.com/146016/65936717-23c2db00-e450-11e9-8b9f-c7c7b489287d.png). Client-side should be aligned with the server-side.
3. Send more precise user role to the analytic system. See comment https://github.com/appirio-tech/connect-app/blob/feature/new-roles/src/actions/loadUser.js#L58-L68

**Comment by vikasrohit** [comment]
@maxceem we can close this one right?

**Comment by maxceem** [comment]
> @maxceem we can close this one right?

No. These are the follow-up tasks after the `2.4.5` release. They mostly come from your comments to the PR https://github.com/topcoder-platform/tc-project-service/pull/387.

These are improvements that we can make after we added new roles in https://github.com/appirio-tech/connect-app/issues/3298.

**Comment by vikasrohit** [comment]
Ohk, I saw that one PR is referencing this issue, so assumed that these changes are done. :)
